### PR TITLE
[master] - Ledger summaries for parent and subaccounts

### PR DESCRIPTION
**Pull request by mk1s**
* Added new MODB view, `ledgers/summary_by_account` to return parent and subaccount ledger summaries.

* Added new crossbar endpoint to `cb_ledgers` to return current modb ledger summary from `../ledgers/summary/{YYYYMM}`.

* Example Payload

```
{
  "page_size": 1,
  "start_key": "g24FAO1KPdYO",
  "data": {
    "summary": {
      "rollovers": {
        "amount": 6.2329,
        "usage": {
          "quantity": 0
        }
      },
      "prorations": {
        "amount": -5.1309,
        "usage": {
          "quantity": 0
        }
      },
      "payments": {
        "amount": 35.0,
        "usage": {
          "quantity": 0
        }
      }
    },
    "account_summary": [
      {
        "cc580f94d7da53816a94b87b2a1d25f8": {
          "account": {
            "id": "cc580f94d7da53816a94b87b2a1d25f8",
            "name": "Account 1"
          },
          "ledgers": {
            "payments": {
              "amount": 35.0,
              "usage": {
                "quantity": 0
              }
            },
            "prorations": {
              "amount": -4.0649,
              "usage": {
                "quantity": 0
              }
            },
            "rollovers": {
              "amount": 6.2329,
              "usage": {
                "quantity": 0
              }
            }
          },
          "total": 37.168
        },
        "de6fd29a54407cfe46e6c9d3828ab0d8": {
          "account": {
            "id": "de6fd29a54407cfe46e6c9d3828ab0d8",
            "name": "Account A"
          },
          "ledgers": {
            "prorations": {
              "amount": -1.066,
              "usage": {
                "quantity": 0
              }
            }
          },
          "total": -1.066
        }
      }
    ]
  },
  "revision": "98804f3d59185d64a59ff7f9e47d1ae9",
  "timestamp": "2019-04-30T22:40:45Z",
  "version": "4.3.1",
  "node": "0ozIxbX-rup7AnGvNSM8JQ",
  "request_id": "051ff47be34fbdf6cb14ed08f4af8415",
  "status": "success",
  "auth_token": "Y2M1ODBmOTRkN2RhNTM4MTZhOTRiODdiMmExZDI1Zjg6NzNlZmY2Mzg2Y2UyMDkxYjVjYTcwMmZjMDA3ZTFkYTk="
}
```

### CODE REVIEW

**Comment by k-anderson** [comment]
@mk1s I think we should clean up the payload, looks like all of my suggestions (including the conflicting ones) got put in ;)

How about:

```
{
  "page_size": 1,
  "start_key": "g24FAO1KPdYO",
  "data": {
    "summary": {
      "rollovers": {
        "amount": 6.2329,
        "usage": {
          "quantity": 0
        }
      },
      "prorations": {
        "amount": -5.1309,
        "usage": {
          "quantity": 0
        }
      },
      "payments": {
        "amount": 35,
        "usage": {
          "quantity": 0
        }
      }
    },
    "breakdown": [
      {
        "account": {
          "id": "cc580f94d7da53816a94b87b2a1d25f8",
          "name": "Account 1"
        },
        "ledgers": {
          "payments": {
            "amount": 35,
            "usage": {
              "quantity": 0
            }
          },
          "prorations": {
            "amount": -4.0649,
            "usage": {
              "quantity": 0
            }
          },
          "rollovers": {
            "amount": 6.2329,
            "usage": {
              "quantity": 0
            }
          }
        },
        "total": 37.168
      },
      {
        "account": {
          "id": "de6fd29a54407cfe46e6c9d3828ab0d8",
          "name": "Account A"
        },
        "ledgers": {
          "prorations": {
            "amount": -1.066,
            "usage": {
              "quantity": 0
            }
          }
        },
        "total": -1.066
      }
    ]
  },
  "revision": "98804f3d59185d64a59ff7f9e47d1ae9",
  "timestamp": "2019-04-30T22:40:45Z",
  "version": "4.3.1",
  "node": "0ozIxbX-rup7AnGvNSM8JQ",
  "request_id": "051ff47be34fbdf6cb14ed08f4af8415",
  "status": "success",
  "auth_token": "Y2M1ODBmOTRkN2RhNTM4MTZhOTRiODdiMmExZDI1Zjg6NzNlZmY2Mzg2Y2UyMDkxYjVjYTcwMmZjMDA3ZTFkYTk="
}
```

File: applications/crossbar/src/modules/cb_ledgers.erl [hunk 1]
```diff
@@ -187,6 +180,19 @@ validate(Context, SourceService) ->
     crossbar_view:load_modb(Context, ?VIEW_BY_SOURCE, ViewOptions).
 
 -spec validate(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+validate(Context, ?SUMMARY, ModbSuffix) ->
+    case ?CURRENT_MODB of
```
Comment (icehess) [inline applications/crossbar/src/modules/cb_ledgers.erl]: What if I want to get the summary about past months? pattern match to check if `ModbSuffix` is in format of YYYYMM and use `kazoo_mdb:get_modb(cb_context:account_id(Context), Year, Month)` to get the correct modb.

File: applications/crossbar/src/modules/cb_ledgers.erl [hunk 2]
```diff
@@ -187,6 +180,19 @@ validate(Context, SourceService) ->
...
+        ModbSuffix ->
+            Summary = summary(Context),
```
Comment (icehess) [inline applications/crossbar/src/modules/cb_ledgers.erl]: Check `resp_status` of both summary and account_summary are `success`.

File: applications/crossbar/src/modules/cb_ledgers.erl [hunk 3]
```diff
@@ -267,6 +273,48 @@ put(Context, Action) ->
...
+%% @end
+%%------------------------------------------------------------------------------
+-spec account_summary(cb_context:context()) ->cb_context:context().
+account_summary(Context) ->
+    Options = [{'group', 'true'}
+              ,{'group_level', 3}
+              ,{'reduce', 'true'}
+              ,{'unchunkable', 'true'}
+              ],
+    Results = kz_datamgr:get_results(cb_context:account_modb(Context), ?VIEW_BY_ACCOUNT, Options),
```
Comment (icehess) [inline applications/crossbar/src/modules/cb_ledgers.erl]: Use `crossbar_view` just like `summary` above(`unchunkable` is crossbar_view option btw :)

**Pull request merged**